lib/serializer.ts

```diff
@@ -618,6 +622,27 @@ function deserializeCompositeType(serializer: Serializer, mapper: CompositeMappe
         instance[responsePropName] = serializer.deserialize(additionalPropertiesMapper, responseBody[responsePropName], objectName + '["' + responsePropName + '"]');
       }
     }
+  } else {
+    const containsSerializedName = (modelProperties: {[propertyName: string]: Mapper}, serializedKey: string): boolean => {
+      for (const key of Object.keys(modelProperties)) {
+        const mapper: Mapper = modelProperties[key];
+        if (mapper.serializedName === serializedKey) {
+          return true;
+        }
+      }
+
+      return false;
+    };
+
+    const isSpecialProperty = (propertyName: string): boolean => {
+      return [ "$", "_" ].includes(propertyName);
```

Comment:
```ts
const isSpecialProperty = (propertyName: string): boolean => 
  [ "$", "_"].includes(propertyName)
```

Reply:
This is specific to the XML scenario, so maybe rename this to `isSpecialXMLProperty`. I'd also recommend changing it to use the normal function syntax (`function isSpecialXMLProperty(propertyName: string): boolean =>...`).